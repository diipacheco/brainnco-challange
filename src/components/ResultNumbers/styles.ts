import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;

  margin-top: 13px;

  @media screen and (min-width: 700px) {
    margin: 0;
  }
`;

export const NumbersList = styled.ul`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  width: 100%;
  list-style: none;

  > li {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 8px 8px;

    width: 44px;
    height: 44px;

    @media screen and (min-width: 700px) {
      width: 111px;
      height: 111px;
      margin: 10px 10px;

      font-size: 26px;
    }

    background: var(--white);
    border-radius: 50%;

    font-weight: 700;
    font-size: 20px;
  }
`;
