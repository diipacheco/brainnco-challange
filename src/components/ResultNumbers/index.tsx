import React, { useEffect } from 'react';
import { useLotteries } from '../../hooks/lotteries';

import { Container, NumbersList } from './styles';

const ResultNumbers: React.FC = () => {
  const { contest, fetchContest, selectedContest } = useLotteries();

  useEffect(() => {
    fetchContest(selectedContest.concursoId);
  }, [fetchContest, selectedContest]);

  return (
    <Container>
      <NumbersList>
        {contest?.numeros?.map(numbers => (
          <li data-testid="result-number-list" key={numbers}>
            {numbers}
          </li>
        ))}
      </NumbersList>
    </Container>
  );
};

export default ResultNumbers;
