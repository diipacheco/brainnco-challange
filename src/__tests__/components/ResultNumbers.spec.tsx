import { render } from '@testing-library/react';

import ResultNumbers from '../../components/ResultNumbers';

jest.mock('../../hooks/lotteries', () => ({
  useLotteries: () => ({
    contest: {
      id: '2359',
      loteria: 0,
      numeros: ['31', '32', '39', '42', '43', '51'],
      data: '2021-08-26T00:42:05.593Z',
    },
    fetchContest: jest.fn(),
    selectedContest: {
      loteriaId: 0,
      concursoId: 2359,
    },
  }),
}));

describe('ResultNumbers', () => {
  it('should match snapshots', () => {
    const wrapper = render(<ResultNumbers />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to render all result numbers of the selected lottery contest', () => {
    const { getAllByTestId } = render(<ResultNumbers />);

    const list = getAllByTestId('result-number-list');

    expect(list).toHaveLength(6);
  });
});
