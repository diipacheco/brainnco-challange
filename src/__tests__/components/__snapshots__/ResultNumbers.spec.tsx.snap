// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultNumbers should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 13px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  list-style: none;
}

.c1 > li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px 8px;
  width: 44px;
  height: 44px;
  background: var(--white);
  border-radius: 50%;
  font-weight: 700;
  font-size: 20px;
}

@media screen and (min-width:700px) {
  .c0 {
    margin: 0;
  }
}

@media screen and (min-width:700px) {
  .c1 > li {
    width: 111px;
    height: 111px;
    margin: 10px 10px;
    font-size: 26px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <ul
          class="c1"
        >
          <li
            data-testid="result-number-list"
          >
            31
          </li>
          <li
            data-testid="result-number-list"
          >
            32
          </li>
          <li
            data-testid="result-number-list"
          >
            39
          </li>
          <li
            data-testid="result-number-list"
          >
            42
          </li>
          <li
            data-testid="result-number-list"
          >
            43
          </li>
          <li
            data-testid="result-number-list"
          >
            51
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM fkYXbU"
    >
      <ul
        class="sc-gtsrHT juOHwc"
      >
        <li
          data-testid="result-number-list"
        >
          31
        </li>
        <li
          data-testid="result-number-list"
        >
          32
        </li>
        <li
          data-testid="result-number-list"
        >
          39
        </li>
        <li
          data-testid="result-number-list"
        >
          42
        </li>
        <li
          data-testid="result-number-list"
        >
          43
        </li>
        <li
          data-testid="result-number-list"
        >
          51
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
